app/ante/ante.go

```diff
@@ -145,3 +145,41 @@ func Recover(logger tmlog.Logger, err *error) {
 		}
 	}
 }
+
+// IsSystemTx determines whether tx is a system tx that's signed by an authorized signer
+// system tx are special types of txs (see in the switch below), or such txs wrapped inside a MsgExec
+// the parameter isAuthorizedSigner is a caller specified function that determines whether the signer of
+// the tx is authorized.
+func IsSystemTx(tx sdk.Tx, isAuthorizedSigner func(string) bool) bool {
```

Comment:
Can we add unit tests for this method?

Reply:
done e032572cc841aa40416a0f9c4a0a7ac7b56d7bfc